internal/config/config_test.go

```diff
@@ -48,6 +48,25 @@ func TestRead(t *testing.T) {
 			},
 			wantErr: nil,
 		},
+		{
+			name: "secrets from file",
+			env: []string{
+				"HCLOUD_TOKEN", "file:/etc/hetzner/token",
+				"ROBOT_USER", "file:/etc/hetzner/user",
+				"ROBOT_PASSWORD", "file:/etc/hetzner/password",
+			},
+			want: HCCMConfiguration{
+				HCloudClient: HCloudClientConfiguration{Token: ""},
+				Robot:        RobotConfiguration{User: "", Password: "", CacheTimeout: 0},
+				Metrics:      MetricsConfiguration{Enabled: false},
+				Instance:     InstanceConfiguration{},
+				LoadBalancer: LoadBalancerConfiguration{Enabled: false},
+				Route:        RouteConfiguration{Enabled: false},
+			},
+			wantErr: errors.New(`failed to read HCLOUD_TOKEN from file: open /etc/hetzner/token: no such file or directory
+failed to read ROBOT_USER from file: open /etc/hetzner/user: no such file or directory
+failed to read ROBOT_PASSWORD from file: open /etc/hetzner/password: no such file or directory`),
+		},
```

Comment:
Could we test the happy path as well?

Reply:
I have not yet covered the positive case in the test cases, as I do not know how you deal with file creation in tests. To run the test for the positive case, a file (e.g. `/tmp/hcloud/token`) would have to be created first.

Reply:
Maybe `internal/testsupport` could be appended to allow tmp file creation.